homeassistant/components/sensor/filter.py

```diff
@@ -44,24 +49,37 @@
 ICON = 'mdi:chart-line-variant'
 
 FILTER_SCHEMA = vol.Schema({
-    vol.Optional(CONF_FILTER_WINDOW_SIZE,
-                 default=DEFAULT_WINDOW_SIZE): vol.Coerce(int),
     vol.Optional(CONF_FILTER_PRECISION,
                  default=DEFAULT_PRECISION): vol.Coerce(int),
 })
 
+# pylint: disable=redefined-builtin
 FILTER_OUTLIER_SCHEMA = FILTER_SCHEMA.extend({
     vol.Required(CONF_FILTER_NAME): FILTER_NAME_OUTLIER,
+    vol.Optional(CONF_FILTER_WINDOW_SIZE,
+                 default=DEFAULT_WINDOW_SIZE): vol.Coerce(int),
     vol.Optional(CONF_FILTER_RADIUS,
                  default=DEFAULT_FILTER_RADIUS): vol.Coerce(float),
 })
 
 FILTER_LOWPASS_SCHEMA = FILTER_SCHEMA.extend({
     vol.Required(CONF_FILTER_NAME): FILTER_NAME_LOWPASS,
+    vol.Optional(CONF_FILTER_WINDOW_SIZE,
+                 default=DEFAULT_WINDOW_SIZE): vol.Coerce(int),
     vol.Optional(CONF_FILTER_TIME_CONSTANT,
                  default=DEFAULT_FILTER_TIME_CONSTANT): vol.Coerce(int),
 })
 
+FILTER_TIME_SMA_SCHEMA = FILTER_SCHEMA.extend({
+    vol.Required(CONF_FILTER_NAME): FILTER_NAME_TIME_SMA,
+    vol.Optional(CONF_TIME_SMA_TYPE,
+                 default=TIME_SMA_LAST): vol.In(
+                     [None, TIME_SMA_LAST]),
```

Comment:
Why is `None` required here? I mean we have a default. The only configuration that I believe would spit out `None` here would be the following:

```yaml
- platform: filter
  # ...
  filters:
    - filter: time_simple_moving_average
      type:
```

And I think we should break when the user writes that.

Reply:
Oops, Otto is right.

Reply:
You are absolutely right! Just created a new PR #13320